kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/LogCompactionWriterSuite.scala

```diff
@@ -0,0 +1,281 @@
+      compactedPath: String,
+      engine: Engine): Seq[Row] = {
+    val fileStatus = FileStatus.of(compactedPath, 0, 0)
+    val batches = engine
+      .getJsonHandler()
+      .readJsonFiles(
+        singletonCloseableIterator(fileStatus),
+        COMPACTED_SCHEMA,
+        Optional.empty())
+    val resBuilder = Seq.newBuilder[Row]
+    while (batches.hasNext()) {
+      val batch = batches.next()
+      val rows = batch.getRows()
+      while (rows.hasNext()) {
+        resBuilder += rows.next()
+      }
+    }
+    resBuilder.result()
+  }
+
+  def addDomainMetadata(path: String): Unit = {
+    spark.sql(
+      s"""
+         |ALTER TABLE delta.`$path`
+         |SET TBLPROPERTIES
+         |('${TableFeatureProtocolUtils.propertyKey(DomainMetadataTableFeature)}' = 'enabled')
+         |""".stripMargin)
+
+    val deltaLog = DeltaLog.forTable(spark, path)
+    val domainMetadata = DeltaSparkDomainMetadata("testDomain1", "", false) ::
+      DeltaSparkDomainMetadata("testDomain2", "{\"key1\":\"value1\"", false) :: Nil
+    deltaLog.startTransaction().commit(domainMetadata, Truncate())
+  }
+
+  Seq(false, true).foreach { includeRemoves =>
+    Seq(false, true).foreach { includeDM =>
+      val removesMsg = if (includeRemoves) " and removes" else ""
+      val dmMsg = if (includeDM) ", include multiple PandM and DomainMetadata" else ""
+      test(s"commits containing adds${removesMsg}${dmMsg}") {
+        withTempDirAndEngine { (tablePath, engine) =>
+          addData(tablePath, alternateBetweenAddsAndRemoves = includeRemoves, numberIter = 10)
+          if (includeDM) {
+            addDomainMetadata(tablePath)
+          }
+
+          val expectedLastCommit = if (includeDM) {
+            11 // 0-9 for add/removes + 2 for enable+set DomainMetatdata
+          } else {
+            9 // 0-9 for add/removes
+          }
+
+          val actionsFromCommits =
+            getActionsFromLog(new Path(tablePath), engine, 0, expectedLastCommit)
+
+          val dataPath = new Path(s"file:${tablePath}")
+          val logPath = new Path(s"file:${tablePath}", "_delta_log")
+
+          val hook = new LogCompactionHook(
+            dataPath,
+            logPath,
+            0,
+            expectedLastCommit,
+            0)
+          hook.threadSafeInvoke(engine)
+          val endCommitStr = f"$expectedLastCommit%020d"
+          val compactedPath =
+            tablePath + s"/_delta_log/00000000000000000000.${endCommitStr}.compacted.json"
+          val actionsFromCompacted = getActionsFromCompacted(compactedPath, engine)
+
+          checkAnswer(actionsFromCompacted, actionsFromCommits)
+        }
+      }
+    }
+  }
+
+  Seq(false, true).foreach { includeRemoves =>
+    val testMsgUpdate = if (includeRemoves) " and removes" else ""
+    test(s"Read table with adds$testMsgUpdate") {
+      withTempDirAndEngine { (tablePath, engine) =>
+        addData(tablePath, alternateBetweenAddsAndRemoves = includeRemoves, numberIter = 10)
```

Comment:
are we sure we haven't done any compaction by default here? should we disable it explicitly?

Reply:
oss spark doesn't support writing minor compactions

Reply:
Ah did not realize this is done using Spark. Anyway to make sure if that is added to Spark this test doesn't start writing them?

Reply:
Yeah, I mean if it's added to spark we'd want to disable compaction, but we can't preemptively do that since there's no config to set currently.